pages/home.py

```diff
@@ -117,73 +132,104 @@ def generate_card(title_text, body_text, icon):
             ])
     return card
 
+def generate_barplot(data, x, y, title):
+    fig = px.bar()
+    if data is not None and not data.empty:
+        fig = px.bar(data, x=x, y=y)
+    fig.update_layout(title=title)
+    return fig
 
 @callback(
-    Output('card-users', 'children'),
-    Input('store-uuids', 'data'),
-)
-def update_card_users(store_uuids):
-    number_of_users = store_uuids.get('length') if has_permission('overview_users') else 0
-    card = generate_card("# Users", f"{number_of_users} users", "fa fa-users")
-    return card
-
-
-@callback(
-    Output('card-active-users', 'children'),
-    Input('store-uuids', 'data'),
+    [
+        Output('card-users', 'children'),
+        Output('skeleton-users', 'visible'),
+        Output('card-active-users', 'children'),
+        Output('skeleton-active-users', 'visible'),
+        Output('card-trips', 'children'),
+        Output('skeleton-trips', 'visible'),
+        Output('fig-sign-up-trend', 'figure'),
+        Output('skeleton-sign-up-trend', 'visible'),
+        Output('fig-trips-trend', 'figure'),
+        Output('skeleton-trips-trend', 'visible'),
+    ],
+    [
+        Input('store-uuids', 'data'),
+        Input('store-trips', 'data'),
+        Input('date-picker', 'start_date'),
+        Input('date-picker', 'end_date'),
+        Input('url', 'pathname'),
+        Input('home-page-load', 'children')
+    ]
 )
-def update_card_active_users(store_uuids):
-    uuid_df = pd.DataFrame(store_uuids.get('data'))
+def update_all_components(store_uuids, store_trips, start_date, end_date, pathname, _):
```

Comment:
I am seeing that we used to have multiple, smaller callbacks for each card. Now we have one big callback that handles everything– which is worse from a code tidiness standpoint.
Can you explain the rationale behind this?